packages/builder/src/components/common/LinkedRowSelector.svelte

```diff
@@ -15,14 +15,25 @@
   let rows = []
   let linkedIds = []
 
-  $: linkedIds = (Array.isArray(linkedRows) ? linkedRows : [])?.map(
-    row => row?._id || row
-  )
+  $: fieldValue = getFieldValue(linkedData)
   $: label = label || capitalise(schema.name)
   $: linkedTableId = linkedTableId || schema.tableId
   $: linkedTable = $tables.list.find(table => table._id === linkedTableId)
   $: fetchRows(linkedTableId)
 
+  const getFieldValue = val => {
+    const linkedIds = (Array.isArray(val) ? val : [])?.map(
+      row => row?._id || row
+    )
+    if (schema.relationshipType === "one-to-many") {
+      return linkedIds?.[0]
+    } else if (schema.type === "bb_reference_single") {
```

Comment:
NAB: This deviates from the logic below - just checking this is intended.

I.e. `{:else if schema.relationshipType === "one-to-many" || schema.type === "bb_reference_single"}`

Implies that `linkedIds[0]` should also be used in this `bb_reference_single` case.

Reply:
Niiiiice catch! I cannot remember now why I made this change. It still returns a list of ids and not a single value. I'll get this updated and give it a test.